pkg/proc/gdbserial/gdbserver_conn.go

```diff
@@ -1549,3 +1507,33 @@ func checksum(packet []byte) (sum uint8) {
 	}
 	return sum
 }
+
+// colonSemicolonParser parses a string in the form:
+//
+//	key1:value1;key2:value2;...;keyN:valueN
+type colonSemicolonParser struct {
+	buf        []byte
+	key, value []byte
+}
+
+func (csp *colonSemicolonParser) next() bool {
+	if len(csp.buf) <= 0 {
+		return false
+	}
+	colon := bytes.Index(csp.buf, []byte{':'})
```

Comment:
Slightly shorter and efficient:
```suggestion
	colon := bytes.IndexByte(csp.buf, ':')
```

Reply:
Done.